learning_observer/learning_observer/canvas.py

```diff
@@ -0,0 +1,341 @@
+import os
+import json
+import string
+import requests
+import recordclass
+import configparser
+import aiohttp
+import aiohttp.web
+import pmss
+import yaml
+
+import learning_observer.settings as settings
+import learning_observer.log_event
+import learning_observer.util
+import learning_observer.auth
+import learning_observer.runtime
+
+cache = None
+
+pmss.register_field(
+    name="default_server",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The default server for Canva",
+    required=True
+)
+pmss.register_field(
+    name="client_id",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The client ID for Canva",
+    required=True
+)
+pmss.register_field(
+    name="client_secret",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The client secret for Canva",
+    required=True
+)
+pmss.register_field(
+    name="access_token",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The access token for Canva",
+    required=True
+)
+pmss.register_field(
+    name="refresh_token",
+    type=pmss.pmsstypes.TYPES.string,
+    description="The refresh token for Canva",
+    required=True
+)
+
+class Endpoint(recordclass.make_dataclass("Endpoint", ["name", "remote_url", "doc", "cleaners"], defaults=["", None])):
```

Comment:
I know you mentioned wanting to wait until the Schoology integration, but committing copied code is **bad practice** unless used as an intermediary step to determine what pieces need to be abstracted. 

I suggest taking this pull request, creating a new file called `lms_integration.py` and adding the common methods to it. For items such as this `Endpoint` class, you should wrap it in a function so you can pass in parameters specific to whatever LMS we are connecting to. Reach out if you need some more guidance on this code/file at all. Abstractions can be weird.

This will make adding new LMSs a ton faster.

Reply:
I've created a `lms_integration.py` to abstract out the common/shared LMS parts. Please review